aws_topology/stackstate_checks/aws_topology/resources/dynamodb.py

```diff
@@ -56,37 +56,31 @@ def collect_table_description(self, table_name):
 
     def construct_table_description(self, table_name):
         return {
-            "Table": {
-                "TableName": table_name,
-                "TableArn": self.agent.create_arn("AWS::DynamoDB::Table", self.location_info, resource_id=table_name),
-            }
+            "TableName": table_name,
+            "TableArn": self.agent.create_arn("AWS::DynamoDB::Table", self.location_info, resource_id=table_name),
         }
 
     def collect_table(self, table_name):
         # If data collection fails, construct the ARN
         data = self.collect_table_description(table_name) or self.construct_table_description(table_name)
-        # If ARN creation fails, skip tag collection
-        if data.get("Table", {}).get("TableArn"):
-            tags = self.collect_tags(data["Table"]["TableArn"])
-        else:
-            tags = []
+        tags = self.collect_tags(data.get("TableArn")) or []
         return TableData(table=data, tags=tags)
 
-    @set_required_access_v2("dynamodb:ListTables")
     def collect_tables(self):
         for table in [
             self.collect_table(table_name)
             for table_name in client_array_operation(self.client, "list_tables", "TableNames")
         ]:
             yield table
 
+    @set_required_access_v2("dynamodb:ListTables")
```

Comment:
this access is more valid on `collect_tables` function.

Reply:
It breaks there. Check all other resources, this is intentional. If the collect_tables function fails, we need to stop the entire service collection